tests/reg_tests/commonUtils.py

```diff
@@ -283,3 +283,55 @@ def spanX(val, geo):
         C[i, 0] *= val
 
     geo.restoreCoef(C, axis_key)
+
+
+def getShapeFunc(lidx):
+    """
+    Get shape dictionaries for use with shape function DVs. Common to DVGeometry and MPhys DVGeo tests.
+    Requires local index from DVGeo object and returns shapes.
+    """
+    shape_1 = {}
+    shape_2 = {}
+
+    k_center = 2
+    i_center = 1
+    n_chord = lidx.shape[0]
+
+    for kk in [-1, 0, 1]:
+        if kk == 0:
+            k_weight = 1.0
+        else:
+            k_weight = 0.5
+
+        for ii in range(n_chord):
+            # compute the chord weight. we want the shape to peak at i_center
+            if ii == i_center:
+                i_weight = 1.0
+            elif ii < i_center:
+                # we are ahead of the center point
+                i_weight = ii / i_center
+            else:
+                # we are behind the center point
+                i_weight = (n_chord - ii - 1) / (n_chord - i_center - 1)
+
+            # get the direction vectors with unit length
+            dir_up = np.array([0.0, 1.0, 0.0])
```

Comment:
Are these Cartesian direction vectors? Should they be an input? I'm not sure the +y and -y will always be considered up/down for all topologies.

Disregard if I'm misunderstanding this.